.github/workflows/release.yml

```diff
@@ -1,86 +1,100 @@
 name: Release version
+
 on:
   push:
     tags:
       - 'v*.*.*'
+  schedule:
+    - cron: '0 0 * * *'
+
+env:
+  TAG_NAME: ${{ github.event_name == 'schedule' && 'nightly' || github.ref_name }}
+  IS_NIGHTLY: ${{ github.event_name == 'schedule' }}
+
 jobs:
-  # this job builds the `forge` and `cast` binaries for the specified platforms
-  # in the matrix. The MacOS binaries do not support M1 yet: https://github.com/actions/runner/issues/805
-  build-artifacts:
-    runs-on:  ${{ matrix.os }}
+  release:
+    name: ${{ matrix.job.target }} (${{ matrix.job.os }})
+    runs-on: ${{ matrix.job.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest]
+        job:
+          # The OS is used for the runner
+          # The platform is a generic platform name
+          # The target is used by Cargo
+          # The arch is either 386 or amd64
+          - os: ubuntu-latest
+            platform: linux
+            target: x86_64-unknown-linux-gnu
+            arch: amd64
+          # The macOS binaries do not support M1 yet: https://github.com/actions/runner/issues/805
```

Comment:
Can we somehow use the Rust `--target` flag to release for M1 even if we're not on an M1 runner?

Reply:
I'll investigate

Reply:
Okay, I've added the target based on https://github.com/rust-lang/rust/issues/73908, running the workflow on my fork to check if it works

Reply:
Seems like we can't do it because of secp256k1-sys https://github.com/onbjerg/foundry/runs/4827202897?check_suite_focus=true

Reply:
Ok, seems like they're working on it. Will disable M1 for now: https://github.com/rust-bitcoin/rust-secp256k1/issues/283